vembrane/__init__.py

```diff
@@ -162,12 +172,18 @@ def filter_vcf(
         env["FORMAT"] = Format(idx, formats)
         env["SAMPLES"] = list(record.samples)
 
-        annotations = dict(record.info).get(ann_key, [""])
-        filtered_annotations = [
-            annotation
-            for annotation in annotations
-            if eval_expression(expression, idx, annotation, annotation_keys, env,)
-        ]
+        if expression.filters_annotations():
+            annotations = dict(record.info).get(ann_key, [""])
```

Comment:
Do we need this empty string here? Should not be an empty list enough?

Reply:
Yes, if there's an ann_key in the expression but there's no ANN field in the record, the list comprehension still needs to run at least once

Reply:
But I guess we can restructure code logic here so it isn't as confusing

Reply:
> Yes, if there's an ann_key in the expression but there's no ANN field in the record, the list comprehension still needs to have at least one entry (i.e. expression.evaluate needs to be called once, which would not happen if the list was empty)

Indeed! :)